libiio.pc.cmakein

```diff
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
-includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+libdir=@libdir_for_pc_file@
```

Comment:
what if we do:
```
libdir=@CMAKE_INSTALL_LIBDIR@
includedir=@CMAKE_INSTALL_INCLUDEDIR@
```

and make sure that `CMAKE_INSTALL_xxxx` are absolute paths?
and if they aren't absolute maybe we cal do `CMAKE_INSTALL_LIBDIR=CMAKE_INSTALL_PREFIX/CMAKE_INSTALL_LIBDIR`

the overall `join_paths()` looks nice, but it's a bit more logic than is required to solve the actual problem;
and it looks to me that `CMAKE_INSTALL_LIBDIR` (and includedir) is used in CMakeLists.txt as well;
so there can be a minor inconsistency between the values of  `CMAKE_INSTALL_LIBDIR`  use in libiio.pc.cmakein  and the ones used in CMakeLists.txt

so, maybe something like?:
```
if(NOT IS_ABSOLUTE "${CMAKE_INSTALL_LIBDIR}")
CMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}
endif
```

Reply:
> and make sure that `CMAKE_INSTALL_xxxx` are absolute paths?

There are `FULL` variants of the variables, see the [docs](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html#result-variables) so that could be used if we decide to go that way.

The reason why there is `${prefix}` is to make it overridable (at least in the cases when it is not absolute), see https://www.bassi.io/articles/2018/03/15/pkg-config-and-paths/.

Reply:
hmm, you are right;
i am just a bit hesitant about having an extra level of indirection for changing from `includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@` to constructing `${prefix}/@CMAKE_INSTALL_INCLUDEDIR@` in CMake via some function;

if ` libiio.pc.cmakein` will only be generated from CMake, then this code seems a bit much;
i'm not fully against it, i just prefer something simpler to read/review

Reply:
I guess the absolute paths are fine if libiio does not expect other software to ever install to their `libdir` or `includedir`.

Reply:
if this would fix your case as well, i would prefer to do it simpler for now;
if we need to add the complexity of `join_paths()` we can add it later;

Reply:
Depends on whether `libiio` is extensible and the extension points are the installation paths. If not, I was not able to come up with a reason https://github.com/analogdevicesinc/libiio/pull/564 should fail.

Reply:
oh; i did not notice the other PR;
too many emails i think